pkg/apis/core/v1beta1/validation/validation.go

```diff
@@ -155,59 +157,108 @@ func ValidateKubeFedConfig(kubeFedConfig *v1beta1.KubeFedConfig) field.ErrorList
-			continue
+	if gates == nil {
+		allErrs = append(allErrs, field.Required(gatesPath, ""))
+	} else {
+		existingNames := make(map[string]bool)
+		for _, gate := range gates {
+			_, ok := existingNames[gate.Name]
+			if ok {
+				allErrs = append(allErrs, field.Duplicate(gatesPath.Child("name"), gate.Name))
+				continue
+			}
+			existingNames[gate.Name] = true
+
+			allErrs = append(allErrs, validateEnumStrings(gatesPath.Child("name"), string(gate.Name),
+				[]string{string(features.PushReconciler), string(features.SchedulerPreferences),
+					string(features.CrossClusterServiceDiscovery), string(features.FederatedIngress)})...)
+
+			allErrs = append(allErrs, validateEnumStrings(gatesPath.Child("configuration"), string(gate.Configuration),
+				[]string{string(v1beta1.ConfigurationEnabled), string(v1beta1.ConfigurationDisabled)})...)
 		}
-		existingNames[gate.Name] = true
-
-		allErrs = append(allErrs, validateEnumStrings(gatesPath.Child("name"), string(gate.Name),
-			[]string{string(features.PushReconciler), string(features.SchedulerPreferences),
-				string(features.CrossClusterServiceDiscovery), string(features.FederatedIngress)})...)
-
-		allErrs = append(allErrs, validateEnumStrings(gatesPath.Child("configuration"), string(gate.Configuration),
-			[]string{string(v1beta1.ConfigurationEnabled), string(v1beta1.ConfigurationDisabled)})...)
 	}
 
 	health := spec.ClusterHealthCheck
 	healthPath := specPath.Child("clusterHealthCheck")
-	allErrs = append(allErrs, validateGreaterThan0(healthPath.Child("periodSeconds"), *health.PeriodSeconds)...)
-	allErrs = append(allErrs, validateGreaterThan0(healthPath.Child("failureThreshold"), *health.FailureThreshold)...)
-	allErrs = append(allErrs, validateGreaterThan0(healthPath.Child("successThreshold"), *health.SuccessThreshold)...)
-	allErrs = append(allErrs, validateGreaterThan0(healthPath.Child("timeoutSeconds"), *health.TimeoutSeconds)...)
+	if health == nil {
+		allErrs = append(allErrs, field.Required(healthPath, ""))
+	} else {
+		allErrs = append(allErrs, validateIntPtrGreaterThan0(healthPath.Child("periodSeconds"), health.PeriodSeconds)...)
+		allErrs = append(allErrs, validateIntPtrGreaterThan0(healthPath.Child("failureThreshold"), health.FailureThreshold)...)
+		allErrs = append(allErrs, validateIntPtrGreaterThan0(healthPath.Child("successThreshold"), health.SuccessThreshold)...)
+		allErrs = append(allErrs, validateIntPtrGreaterThan0(healthPath.Child("timeoutSeconds"), health.TimeoutSeconds)...)
+	}
 
 	sync := spec.SyncController
 	syncPath := specPath.Child("syncController")
-	allErrs = append(allErrs, validateEnumStrings(syncPath.Child("adoptResources"), string(*sync.AdoptResources),
-		[]string{string(v1beta1.AdoptResourcesEnabled), string(v1beta1.AdoptResourcesDisabled)})...)
+	if sync == nil {
+		allErrs = append(allErrs, field.Required(syncPath, ""))
+	} else {
+		allErrs = append(allErrs, validateEnumStrings(syncPath.Child("adoptResources"), string(*sync.AdoptResources),
```

Comment:
It will be crash if `sync.AdoptResources` is `nil`. We had a test case to trigger it though currently it is wrongly skipped.  We should also check if `sync.AdoptResources` is `nil` or not because access it.

Reply:
Done.